src/lib/optimizer/optimizer.cpp

```diff
@@ -216,7 +217,9 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
   // beneficial. However, TPC-H Q 21 (that is already long-running) degrades drastically. See:
   // https://github.com/hyrise/hyrise/pull/2536#issuecomment-1423076256
   // TODO(anyone): Re-evaluate this in the future.
-  optimizer->add_rule(std::make_unique<SemiJoinReductionRule>());
+  // optimizer->add_rule(std::make_unique<SemiJoinReductionRule>());
+
+  optimizer->add_rule(std::make_unique<DataInducedPredicateRule>());
```

Comment:
These two rules are basically competing with each other, applying similar optimizations but in different ways. So they can't be commented in both, but I agree that we should reset it to the state in which the SJR is applied. Thanks for the pointer!

Reply:
For a master-mergeable rule, we need to combine both rules. However, since that was not possible in the scope of the seminar, I think it's a good starting point for measurements w/ only one rule.